### PR TITLE
use fs-extra to save files to disk

### PR DESCRIPTION
**Pull request by hakubo**
it can ensure that all dirs in path are created so this change:
- fixes https://github.com/Cropster/ember-visual-test/issues/5
- adds support for sub directories for images, so they can be grouped together

TODO: 

- [x] update documentaton
- [x] add a test with more sub directories

@mydea does it sound good? would this be accepted?

### CODE REVIEW

**Comment by scottkidder** [comment]
Looking at why the tests failed in travis and it seems to be related to font rendering changing somehow.

File: index.js [hunk 1]
```diff
@@ -289,11 +288,12 @@ module.exports = {
       this._makeScreenshots(url, fileName, { selector, fullPage, delayMs }).then(({ newBaseline, newScreenshotUrl }) => {
         data.newScreenshotUrl = newScreenshotUrl;
         data.newBaseline = newBaseline;
+
         return this._compareImages(fileName);
       }).then(() => {
         data.status = 'SUCCESS';
         res.send(data);
-      }).catch((reason) => {
+        }).catch((reason) => {
```
Comment (mydea) [inline index.js]: This indentation looks wrong to me, could you revert it?

**Comment by mydea** [comment]
Thanks for that, that looks pretty neat!
However, it seems that the rendering on Linux is not always consistent - probably due to different Linux distros or other environment factors. For now, could you just remove the linux-... baseline images, so that the tests pass - or replace them with the ones from the travis run (you can find the urls in here: https://travis-ci.org/Cropster/ember-visual-test/jobs/383409735)?

We should then maybe think about making this even more robust, e.g. splitting it up not just by os, but also by ci runs. (e.g. linux-..., travis-...)

**Comment by hakubo** [comment]
@mydea replaced baseline with the ones from travis, also updated docs. Anything else?

**Comment by mydea** [comment]
LGTM, thank you :)

**Pull request merged**